homeassistant/helpers/config_validation.py

```diff
@@ -89,14 +89,42 @@ def validate(obj: Dict) -> Dict:
 
 def boolean(value: Any) -> bool:
     """Validate and coerce a boolean value."""
+    if value is None:
+        return False
+    if isinstance(value, bool):
+        return value
     if isinstance(value, str):
-        value = value.lower()
+        value = value.lower().strip()
         if value in ('1', 'true', 'yes', 'on', 'enable'):
             return True
         if value in ('0', 'false', 'no', 'off', 'disable'):
             return False
         raise vol.Invalid('invalid boolean value {}'.format(value))
-    return bool(value)
+    if isinstance(value, Number):
+        return value != 0
+    raise vol.Invalid('invalid boolean value {}'.format(value))
+
+
+def boolean_true(value: Any) -> bool:
+    """Coerce a boolean value to true, or return false."""
+    if isinstance(value, bool):
+        return value
+    if isinstance(value, str):
+        value = value.lower().strip()
+        return value in ('1', 'true', 'yes', 'on', 'enable')
+    if isinstance(value, Number):
+        return value != 0
+    return False
+
+
+_WS = re.compile('\\s*')
+
+
+def whitespace(value: Any) -> str:
```

Comment:
This should really be called `strip_whitespace`, but I first want to see a use case and an application of this. We cant just add validators that seem useful, without using them.

Reply:
Use case is really using with the Or validation, for None or some kind of object.

```vol.Or(cv.whitespace, vol.integer)```

Reply:
but where in the code is it used? Because as far as I know, we don't run output of templates through validators? Instead, we run config through it.

Reply:
> Because as far as I know, we don't run output of templates through validators?

We will soon :grin:. I'm planning to validate the rendered outputs of all the templates in the template component, because it makes a lot of sense to do so. Then you can log errors on invalid output.

You can't know what a template is going to produce before you run it (halting problem), but you can validate the output.

Reply:
How about calling it `whitespace_none_or` and passing in the chained validator? That's really the main use case.

Reply:
Or even `whitespace_as` and supply a value to use, but still use it with the `vol.Or` validator.

Then you can have useful stuff like `vol.Or(cv.whitespace_as([]), [cv.string])`

Reply:
I just don't know if this needs to be a validator that raises invalid? And in your example you have a list or a single value. What if the list contains a whitespace value?